src/external/ACSAdapter/AMSFileScanner.ts

```diff
@@ -0,0 +1,138 @@
+    next?: any; // eslint-disable-line @typescript-eslint/no-explicit-any
+    activity?: any; // eslint-disable-line @typescript-eslint/no-explicit-any
+}
+
+class AMSFileScanner {
+    public amsClient: FramedClient;
+    public scanResults: Map<string, FileScanResult> | null = null;
+    public interval: number = defaultScanInterval;
+    private shouldQueueScan: boolean;
+
+    constructor(amsClient: FramedClient, interval: number = defaultScanInterval) {
+        this.amsClient = amsClient;
+        this.scanResults = new Map<string, FileScanResult>();
+        this.interval = interval;
+        this.shouldQueueScan = false;
+        this.queueScan();
+    }
+
+    public async queueScan(): Promise<void> {
+        this.shouldQueueScan = true;
+
+        try {
+            await this.scanFiles();
+        } catch (e) {
+            console.error(e);
+        } finally {
+            await sleep(this.interval);
+            this.shouldQueueScan && await this.queueScan();
+        }
+    }
+
+    public retrieveFileScanResult(id: string): FileScanResult | undefined {
+        return this.scanResults?.get(id);
+    }
+
+    public addOrUpdateFile(id: string, fileMetadata: FileMetadata, scan: FileScanResponse): void {
+        const scanResult = this.retrieveFileScanResult(id);
+        this.scanResults?.set(id, {
+            ...(scanResult || {}),
+            fileMetadata,
+            scan
+        });
+    }
+
+    public addNext(id: string, next: Function): void {
+        const scanResult = this.scanResults?.get(id);
+        if (scanResult) {
+            this.scanResults?.set(id, {...scanResult, next});
+        }
+    }
+
+    public addActivity(id: string, activity: any): void {  // eslint-disable-line @typescript-eslint/no-explicit-any, @typescript-eslint/explicit-module-boundary-types
+        const scanResult = this.scanResults?.get(id);
+        if (scanResult) {
+            this.scanResults?.set(id, {...scanResult, activity});
+        }
+    }
+
+    public async scanFileCallback(scanResult: FileScanResult, id: string): Promise<void> {
+        const {fileMetadata, next, activity} = scanResult;
+
+        if (scanResult?.scan?.status === AMSDownloadStatus.INPROGRESS) {
+            try {
+                const response: any = await this.amsClient.getViewStatus(fileMetadata); // eslint-disable-line @typescript-eslint/no-explicit-any
+                const {view_location, scan} = response;
+
+                this.addOrUpdateFile(id, scanResult.fileMetadata, scan);
+
+                if (scan.status === AMSDownloadStatus.PASSED && next && activity) {
+                    let blob: any; // eslint-disable-line @typescript-eslint/no-explicit-any
+                    try {
+                        blob = await this.amsClient.getView(fileMetadata, view_location); // eslint-disable-line @typescript-eslint/no-explicit-any
+                    } catch (error) {
+                        console.error(error);
+                    }

```

Comment:
why is wraped blob assignation?, if it fails then it will just print the error message, but it doesnt stop the logic, we still create the file with an empty blob, 

is that correct?, 

could we extract blob assignation to its own function, to reduce verbosity in this method?

Reply:
It's a guard in case the API to AMS fails. If it fails, the File would be returned with no content.